### ISSUE TITLE
Saving notes fails in isolate

### ISSUE THREAD

**Issue by ZebraVogel94349** [saber-notes/saber#1031]
### Describe the bug

Saving a note with lots of images sometimes fails, resulting in the note to be corrupted. I think this is caused by #1017, because this error occurs:

```
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: LateInitializationError: Field 'documentsDirectory' has not been initialized.
#0      FileManager.documentsDirectory (package:saber/data/file_manager/file_manager.dart)
#1      FileManager.getFile (package:saber/data/file_manager/file_manager.dart:115:13)
#2      new PngEditorImage.fromJson (package:saber/components/canvas/image/png_editor_image.dart:57:33)
#3      new EditorImage.fromJson (package:saber/components/canvas/image/editor_image.dart:121:29)
#4      EditorPage.parseImageJson (package:saber/data/editor/page.dart:198:21)
#5      new EditorPage.fromJson (package:saber/data/editor/page.dart:104:13)
#6      EditorCoreInfo._parsePagesJson.<anonymous closure> (package:saber/data/editor/editor_core_info.dart:227:43)
#7      MappedListIterable.elementAt (dart:_internal/iterable.dart:425:31)
#8      ListIterator.moveNext (dart:_internal/iterable.dart:354:26)
#9      new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
#10     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
#11     new List.of (dart:core-patch/array_patch.dart:39:18)
#12     ListIterable.toList (dart:_internal/iterable.dart:224:7)
#13     EditorCoreInfo._parsePagesJson (package:saber/data/editor/editor_core_info.dart:235:10)
#14     new EditorCoreInfo.fromJson (package:saber/data/editor/editor_core_info.dart:164:14)
#15     EditorCoreInfo._loadFromFileIsolate (package:saber/data/editor/editor_core_info.dart:420:29)
#16     EditorCoreInfo.loadFromFileContents.isolate (package:saber/data/editor/editor_core_info.dart:358:35)
#17     WorkerImpl._anotherIsolate.<anonymous closure> (package:worker_manager/src/worker/worker_io.dart:115:33)
#18     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
```

### To reproduce

1. Create a note, import some images, write a few pages
2. Try to save the note

### Expected behavior

Save the note normally, without corruption

### Saber version

v.0.18.3

### Device

- Device: Samsung Galaxy Tab S7 FE / PC
- OS: Android 13 / Linux Mint 21.1

**Comment by ZebraVogel94349** [comment]
This issue still exists. When I import any large PDF file (about 20 pages), draw a lot with the pen (a few MB in strokes) and try to save it, I get this error:
`flutter: SEVERE: EditorCoreInfo: Failed to load file from /test: LateInitializationError: Field 'documentsDirectory' has not been initialized.
[ERROR:flutter/runtime/dart_vm_initializer.cc(41)] Unhandled Exception: LateInitializationError: Field 'documentsDirectory' has not been initialized.
#0      FileManager.documentsDirectory (package:saber/data/file_manager/file_manager.dart)
#1      FileManager.getFile (package:saber/data/file_manager/file_manager.dart:115:19)
#2      new PdfEditorImage.fromJson (package:saber/components/canvas/image/pdf_editor_image.dart:52:31)
#3      new EditorImage.fromJson (package:saber/components/canvas/image/editor_image.dart:113:29)
#4      EditorPage.parseImageJson (package:saber/data/editor/page.dart:198:21)
#5      new EditorPage.fromJson (package:saber/data/editor/page.dart:104:13)
#6      EditorCoreInfo._parsePagesJson.<anonymous closure> (package:saber/data/editor/editor_core_info.dart:227:43)
#7      MappedListIterable.elementAt (dart:_internal/iterable.dart:425:31)
#8      ListIterator.moveNext (dart:_internal/iterable.dart:354:26)
#9      new _GrowableList._ofEfficientLengthIterable (dart:core-patch/growable_array.dart:189:27)
#10     new _GrowableList.of (dart:core-patch/growable_array.dart:150:28)
#11     new List.of (dart:core-patch/array_patch.dart:39:18)
#12     ListIterable.toList (dart:_internal/iterable.dart:224:7)
#13     EditorCoreInfo._parsePagesJson (package:saber/data/editor/editor_core_info.dart:235:10)
#14     new EditorCoreInfo.fromJson (package:saber/data/editor/editor_core_info.dart:164:14)
#15     EditorCoreInfo._loadFromFileIsolate (package:saber/data/editor/editor_core_info.dart:420:29)
#16     EditorCoreInfo.loadFromFileContents.isolate (package:saber/data/editor/editor_core_info.dart:358:35)
#17     WorkerImpl._anotherIsolate.<anonymous closure> (package:worker_manager/src/worker/worker_io.dart:115:33)
#18     _RawReceivePort._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)`

**Comment by ZebraVogel94349** [comment]
I just noticed, that sometimes all FileManager Variables are uninitialized/set to the initial value, which shouldn't be possible since they are static and the .init() method which is called in main_common.dart initialized them. I printed `documentsDirectory`, which was set correctly, tried to save an imported PDF file with a lot of annotations and just before saving it was uninitialized. According to [this StackOverflow post](https://stackoverflow.com/questions/45772318/flutter-dart-static-variables-lost-keep-getting-reinitialized), issues like this can be caused by mixing uppercase and lowercase letters in import statements, but I couldn't find anything like it. It could be a bug in flutter.
